src/Sylius/Component/Core/Promotion/Action/UnitDiscountPromotionActionCommand.php

```diff
@@ -107,4 +111,23 @@ private function calculate(int $unitTotal, ?int $minimumPrice, int $promotionAmo
 
         return $promotionAmount;
     }
+
+    private function canPromotionBeApplied(OrderItemUnitInterface $unit, PromotionInterface $promotion): bool
```

Comment:
Shouldn't we make it `protected`? It's a solid part of logic, that would not be easily replaceable (as is the private function in the extendable class)